FtpClient/FtpCli/InitializeSession.cs

```diff
@@ -0,0 +1,71 @@
+using System;
+
+namespace FtpCli
+{
+  // The IntializeSession class prompts the user for session details
+  // which are stored as a Data object
+  public class InitializeSession
+  {
+      Data data;
+
+      public InitializeSession() {
+        data = new Data();
+      }
+
+    // prompt user for server, username, and commands (if provided)
+    public Data initialPrompt(string[] args) {
+      string response = "";
+
+      do {
+        response = this.promptUser("Enter Server: ");
+      } while (!validateNonEmptyResponse(response));
+      this.data.setServer(response);
+
+      do {
+        response = this.promptUser("Enter Username: ");
+      } while (!validateNonEmptyResponse(response));
+      this.data.setUser(response);
+
+      response = this.promptUser("Enter Command (or `Enter` if none): ");
+      this.data.setCommands(response);
+
+      return this.data;
+    }
+
+    // prompts user and returns input
+    public string promptUser(string prompt) {
+
+      Console.WriteLine(prompt);
+      return Console.ReadLine();
+    }
+
+    // validates that the response contains valid string content
+    public bool validateNonEmptyResponse(string response) 
+    {
+      // incorrect number of values passed in
+      if (response == "") {
+        return false;
+      }
+      string[] responseAsArray = response.Split(" ");
+      if (responseAsArray.Length > 1) {
+        return false;
+      }
+      return true;
+    }
+
+    public void confirmServer() {
```

Comment:
It might be better to just have the print statements exist in main rather than pushing them into the lower classes. Just have getters and setters on the string args and we can do something like `Console.Write("\nConnecting to " + this.data.getServer() + " with username " + this.data.getUser() + "...");`? What is the consensus here?

Reply:
Makes sense, especially because the confirm functions are used once and we have access to the Data object. I deleted the confirm functions and used your suggestion.